src/ORM/Search/SearchContext.php

```diff
@@ -140,6 +141,13 @@ protected function applyBaseTableFields()
      */
     public function getQuery($searchParams, $sort = false, $limit = false, $existingQuery = null)
     {
+        if (!in_array(gettype($limit), ['array', 'NULL', 'string'])) {
```

Comment:
Just occurred to me that this deprecation notice will fire when not passing in an explicit value for `$limit` which would be super annoying

Can you also add in a check for `count(func_get_args()) >= 3` (I think this is correct) to validate that `$limit` was actually passed in.  Seems that `func_get_args()` won't include default values which is rather handy in this scenario

Reply:
Updated